raiden-ts/src/transport/epics.ts

```diff
@@ -1193,11 +1203,18 @@ function waitMemberAndSend$(
             }),
             pluck('roomId'),
           ),
-    ),
+        ),
+      );
+    }),
     take(1), // use first room/user which meets all requirements/filters above
-    mergeMap((roomId) =>
-      defer(() => matrix.sendEvent(roomId, type, content, '')).pipe(
-        mapTo(roomId),
+    mergeMap((via) =>
+      defer(
+        () =>
+          typeof via === 'string'
+            ? matrix.sendEvent(via, type, content, '') // via room
+            : via.send(content.body), // via RTC channel
+      ).pipe(
+        mapTo(typeof via === 'string' ? via : via.label),
```

Comment:
Maybe better to have something more obvious/self explanatory here, like `Mode.MATRIX ? ... : ...` instead of `typeof via === 'string'`

Reply:
This is an implementation detail on an local/internal helper function, I don't think it's worth the hassle. Notice the result of this observable isn't used anywhere, so all it does is performed as side-effects and the returned type doesn't matter.

Reply:
Any change to add more details to where this detail comes from though? Some kind of comment with any information regarding the context?

Reply:
I added a comment for it. Otherwise, you can just see, all values from the observables returned by this function is rejected, it's like a wrapper for `sendEvent`, but which also may wait for several reactive conditions, and will send the message when possible, and just return anything to notify it succeeded (or errors otherwise)